tests/testutil/router.go

```diff
@@ -31,7 +33,11 @@ func (tr *TestRoute) Add(t *testing.T, domain string, routePath string, action s
 	tr.Path = routePath
 	tr.Name = fmt.Sprintf("%s/%s", testingNamespace, domain)
 	route := fmt.Sprintf("%s.%s%s", domain, testingNamespace, routePath)
-	_, err := RioCmd([]string{"route", "add", route, action, target.Name})
+	opts := wait.Backoff{Factor: 1.0, Steps: 3}
```

Comment:
Does it make sense to just add this to RioCmd ?

Reply:
Actually @cbron I was discussing this one with @rancher-max but I'm kind of looking for an opinion on how to handle this one in the `RioCmd` method because it makes sense. 

But `cmd.CombinedOutput()` returns the stdout and we use that to print errors which we think are important for debugging.

```go
	var stdOutErr []byte
	opts := wait.Backoff{Factor: 1.0, Steps: 3}
	err := retry.RetryOnConflict(opts, func() error {
		stdOutErr, err := cmd.CombinedOutput()
		return err
	})
	if err != nil {
		return "", fmt.Errorf("%s: %s", err.Error(), stdOutErr)
	}
```
So how do we keep `stdOutError` in that case because `RetryOnConflict` only returns the errors after the retries.

Reply:
Hmm true, `RetryOnConflict` func has only an error as return param. You could redo that logic (and `OnError`) in our framework, even though its unfortunate we can't reuse code.

Reply:
In the meantime these were the most "conflicting" areas I identified on new clusters when running the tests in parallel. I can work in a reimplementation and refactor all these in a separate PR. Would you agree?

Reply:
Sure. We can chat next week about approach too, see if there is a better way we aren't thinking of.

Reply:
@rancher-max @cbron Ive added a simple function for retry in testutil. Seems to do the trick.